framework_agreement/model/pricelist.py

```diff
@@ -16,75 +14,419 @@
-        agreement.
+        if self.picking_type_id:
+            pick_type = self.picking_type_id
 
-        If it is a supplier price list, agreement will be taken in account
-        and use the price of the agreement if required.
+            if pick_type.warehouse_id.partner_id:
+                self.dest_address_id = pick_type.warehouse_id.partner_id.id
 
-        If there is not enough available qty on agreement,
-        standard price will be used.
+            if pick_type.default_location_dest_id:
+                self.location_id = pick_type.default_location_dest_id
+                self.related_location_id = pick_type.default_location_dest_id
 
-        This is maybe a faulty design and we should use on_change override
+    def _price_rule_get_multi(self, cr, uid, pricelist,
+                              products_by_qty_by_partner, context=None):
+        """Ugly duplication to implement the boolean use_agreement_prices."""
+        context = context or {}
+        date = context.get('date') or time.strftime('%Y-%m-%d')
 
-        """
-        if context is None:
-            context = {}
-        agreement_obj = self.pool['framework.agreement']
-        res = super(product_pricelist, self).price_get(
-            cr, uid, ids, prod_id, qty, partner=partner, context=context)
-        if not partner:
-            return res
-        for pricelist_id in res:
-            if (pricelist_id == 'item_id' or not
-                    self._plist_is_agreement(cr, uid,
-                                             pricelist_id, context=context)):
-                continue
-            now = datetime.strptime(fields.date.today(),
-                                    DEFAULT_SERVER_DATE_FORMAT)
-            date = context.get('date') or context.get('date_order') or now
-            prod = self.pool['product.product'].browse(cr, uid, prod_id,
-                                                       context=context)
-            agreement = agreement_obj.get_product_agreement(
-                cr, uid,
-                prod.product_tmpl_id.id,
-                partner,
-                date,
-                qty=qty,
-                context=context
-            )
-            if agreement is not None:
-                currency = agreement_obj._get_currency(
-                    cr, uid, partner, pricelist_id,
-                    context=context
-                )
-                res[pricelist_id] = agreement.get_price(qty, currency=currency)
-        return res
+        products = map(lambda x: x[0], products_by_qty_by_partner)
+        currency_obj = self.pool.get('res.currency')
+        product_obj = self.pool.get('product.template')
+        product_uom_obj = self.pool.get('product.uom')
+        price_type_obj = self.pool.get('product.price.type')
+
+        if not products:
+            return {}
+
+        version = False
+        for v in pricelist.version_id:
+            if (
```

Comment:
I will add a function on pricelist.version model. That typically the kind of test you want to override. It will also be more readable